src/tiramisu_core.cpp

```diff
@@ -7810,4 +7811,98 @@ void tiramisu::buffer::tag_gpu_register() {
     set_auto_allocate(false);
 }
 
+std::string get_rank_string_type(tiramisu::rank_t rank_type)
+{
+    if (rank_type == rank_t::r_sender)
+        return "r_sender";
```

Comment:
Can you use shorter names? r_snd and r_rcv? These dimension names will appear a lot in maps and sets so big names will be a problem.